packages/server/modules/shared/authz.ts

```diff
@@ -322,38 +302,50 @@ export const authPipelineCreator = (
   return pipeline
 }
 
-export const streamWritePermissions: AuthPipelineFunction[] = [
-  validateServerRole({ requiredRole: Roles.Server.Guest }),
+export const streamWritePermissionsPipelineFactory = (
+  deps: ValidateRoleBuilderDeps & ValidateRequiredStreamDeps
+): AuthPipelineFunction[] => [
+  validateServerRoleBuilderFactory(deps)({ requiredRole: Roles.Server.Guest }),
   validateScope({ requiredScope: Scopes.Streams.Write }),
-  contextRequiresStream({
-    getStream,
-    getAutomationProject: getAutomationProjectFactory({ db })
-  }),
-  validateStreamRole({ requiredRole: Roles.Stream.Contributor }),
-  validateResourceAccess
-]
-export const streamReadPermissions: AuthPipelineFunction[] = [
-  validateServerRole({ requiredRole: Roles.Server.Guest }),
-  validateScope({ requiredScope: Scopes.Streams.Read }),
-  contextRequiresStream({
-    getStream,
-    getAutomationProject: getAutomationProjectFactory({ db })
-  }),
-  validateStreamRole({ requiredRole: Roles.Stream.Contributor }),
+  validateRequiredStreamFactory(deps),
+  validateStreamRoleBuilderFactory(deps)({ requiredRole: Roles.Stream.Contributor }),
   validateResourceAccess
 ]
 
-if (adminOverrideEnabled()) streamReadPermissions.push(allowForServerAdmins)
+export const streamReadPermissionsPipelineFactory = (
+  deps: ValidateRoleBuilderDeps &
+    ValidateRequiredStreamDeps & {
+      adminOverrideEnabled: typeof adminOverrideEnabled
+    }
+): AuthPipelineFunction[] => {
+  const ret: AuthPipelineFunction[] = [
+    validateServerRoleBuilderFactory(deps)({ requiredRole: Roles.Server.Guest }),
+    validateScope({ requiredScope: Scopes.Streams.Read }),
+    validateRequiredStreamFactory(deps),
+    validateStreamRoleBuilderFactory(deps)({ requiredRole: Roles.Stream.Contributor }),
+    validateResourceAccess
+  ]
+
+  if (deps.adminOverrideEnabled()) ret.push(allowForServerAdmins)
+
+  return ret
+}
+
+export const throwForNotHavingServerRoleFactory =
+  (deps: { validateServerRole: ValidateServerRoleBuilder }): ValidateUserServerRole =>
+  async (context: AuthContext, requiredRole: ServerRoles) => {
+    const { authResult } = await deps.validateServerRole({ requiredRole })({
+      context,
+      authResult: { authorized: false }
+    })
+    if (authHasFailed(authResult))
+      throw authResult.error ?? new Error('Auth failed without an error')
+    return true
+  }
 
-export const throwForNotHavingServerRole = async (
-  context: AuthContext,
-  requiredRole: ServerRoles
-) => {
-  const { authResult } = await validateServerRole({ requiredRole })({
-    context,
-    authResult: { authorized: false }
+// Global singleton for easy access
```

Comment:
Another one of those common singleton utils that get imported all over the place